### ISSUE TITLE
[java] StringToString false-positive with local method name confusion

### ISSUE THREAD

**Issue by adangel** [pmd/pmd#3977]
**Affects PMD Version:** 6.43.0

**Rule:** [StringToString](https://pmd.github.io/latest/pmd_rules_java_performance.html#stringtostring)

**Description:**

When you have a method chain call and at the end the `toString()` call. And there is a local method with the same name as the method call, then this rule is confused. If the local method returns a String, then this rule is triggered.

The rule searches always for local method declarations, but that's wrong - at least for method chains.

**Code Sample demonstrating the issue:**

```java
public class StringToString_FP {
  private String s;

  public void print(A a) {
    this.s = a.getB().getC().toString(); // line 5 - false positive
    this.s = String.valueOf(a.getB().getC()); // workaround
    System.out.println(s);
  }

  public String getC() {
    return "";
  }
}

interface A {
    B getB();
}

interface B {
    Character getC();
}

```

`B::getC` returns a Character, `StringToString_FP::getC` returns a String.
At line 5, `B::getC` is called.

**Expected outcome:**

PMD reports a violation at line 5, but that's wrong. That's a false positive.
